x-pack/test/functional/services/user_menu.js

```diff
@@ -42,8 +42,10 @@ export function UserMenuProvider({ getService }) {
         return;
       }
 
-      await testSubjects.click('userMenuButton');
-      await retry.waitFor('user menu opened', async () => await testSubjects.exists('userMenu'));
+      await retry.try(async () => {
+        await testSubjects.click('userMenuButton');
+        await testSubjects.existOrFail('userMenu');
+      });
```

Comment:
This is a tip from @flash1293 that helped the flakiness a lot when testing locally. It seems like we need to click the button several times to gain focus.